src/background/services/wallet/handlers/eth_sendTransaction/eth_sendTransaction.ts

```diff
@@ -255,38 +261,59 @@ export class EthSendTransactionHandler extends DAppRequestHandler<
         network
       );
 
-      const notificationId = Date.now().toString();
-      await browser.notifications.create(notificationId, {
-        type: 'basic',
-        iconUrl: '../../../../images/icon-32.png',
-        title: 'Confirmed transaction',
-        message: `Transaction confirmed! View on the explorer.`,
-        priority: 2,
-      });
-
-      const openTab = async (id: string) => {
-        if (id === notificationId) {
-          const explorerUrl = getExplorerAddressByNetwork(network, txHash);
-          await browser.tabs.create({ url: explorerUrl });
-        }
-      };
+      const notificationId = crypto.randomUUID();
 
-      browser.notifications.onClicked.addListener(openTab);
-      browser.notifications.onClosed.addListener((id: string) => {
-        if (id === notificationId) {
-          browser.notifications.onClicked.removeListener(openTab);
-        }
-      });
+      await this.#createTxNotification(
+        notificationId,
+        {
+          type: 'basic',
+          iconUrl: '../../../../images/icon-32.png',
+          title: 'Pending transaction',
+          message: `Transaction pending! View on the explorer.`,
+          priority: 2,
+        },
+        network,
+        txHash
+      );
 
-      /*
-        notifications.onClosed is only triggered when a user close the notification.
-        And the notification is automatically closed 5 secs if user does not close it.
-        To mimic onClosed for system, using setTimeout.
-      */
-      setTimeout(async () => {
-        browser.notifications.onClicked.removeListener(openTab);
-        await browser.notifications.clear(notificationId);
-      }, 5000);
+      provider
+        .waitForTransaction(txHash)
+        .then(async (tx) => {
+          const duration = measurement.end();
+
+          const isTxSuccessul = tx?.status === 1;
+          await browser.notifications.clear(notificationId); // close transaction pending notification
+          await this.#createTxNotification(
+            crypto.randomUUID(),
+            {
+              type: 'basic',
+              iconUrl: '../../../../images/icon-32.png',
+              title: isTxSuccessul
+                ? 'Confirmed transaction'
+                : 'Failed transaction',
+              message: `Transaction ${
+                isTxSuccessul ? 'confirmed' : 'failed'
+              }! View on the explorer.`,
+              priority: 2,
+            },
+            network,
+            txHash
+          );
```

Comment:
Should we check if the wallet's locked state and trigger the notification only if it's unlocked? 

```suggestion
          if (!this.lockService.locked) {
            await this.#createTxNotification(
              crypto.randomUUID(),
              {
                type: 'basic',
                iconUrl: '../../../../images/icon-32.png',
                title: isTxSuccessul
                  ? 'Confirmed transaction'
                  : 'Failed transaction',
                message: `Transaction ${
                  isTxSuccessul ? 'confirmed' : 'failed'
                }! View on the explorer.`,
                priority: 2,
              },
              network,
              txHash
            );
          }
```